drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -526,15 +529,37 @@ local function initialize_switch(driver, device)
     device:set_field(COMPONENT_TO_ENDPOINT_MAP_BUTTON, component_map, {persist = true})
   end
 
-  if num_switch_server_eps > 0 then
+  if #switch_eps > 0 and #button_eps > 0 then
+    if tbl_contains(STATIC_BUTTON_PROFILE_SUPPORTED, #button_eps) then
+      local dimmable_light = false
+      for _, ep in ipairs(device.endpoints) do
+        for _, dt in ipairs(ep.device_types) do
+          if dt.device_type_id == DIMMABLE_LIGHT_DEVICE_TYPE_ID then
```

Comment:
This is essentially saying we are only going to support swithc/button combos where the switch component is a dimmable light. We could add handling for the other switch device types pending what devices are on the market. Are dimmable light + buttons the only switch/button combo devices we have tested/know about?

If we do expand this, we will need to create more profiles as I mentioned above because this design couples a switch component with the button components in a single MCD profile.

Reply:
Yes, dimmable light + buttons is the only combination we are aware of so far. The logic here could easily be expanded to cover more device types in the future as needed. And, you are right that new profiles would also be needed in that case.

Reply:
So if a switch/button combo device type we do not support is onboarded, then how would it look?

It looks like this would go down the path of trying to configure the buttons immediately, but we can't know if the device joined to a button or switch profile, so it could start configuring button that don't even exist in the profile. Even if it joins as a button profile, it wouldn't neccessarily have all the buttons represented because the fingerprint defaults to `button-battery`, a single component device.

This is because this wouldn't detect a match for the `DIMMABLE_LIGHT_DEVICE_TYPE_ID` type, which would mean `profile_name` would be `nil`. We should define a default behavior for unsupported devices in which we can provide at least partial support. We'd essentially have to choose between partially supporting the switch half of the device, or the button half. I would probably choose the button half if we are going to provide partial support. If profile name is nil, then I think we need to set the profile name correctly to be just buttons:
```
profile_name = string.format("%d-button", #button_eps)
```

Also one other thing to point out is that if the main endpoint is a switch device type that is unsupported, this logic would also fail. So if we commit to just supporting the button half of unsupported combos, then I'd add an additional check at that top of this function to confirm that we only handle the switch as the main endpoint if we plan to support it. Does that makes sense?

Reply:
In 25cf9d3d30e4bfd062ee4b21b26303b624bc0682 I updated `find_default_endpoint` to use the first switch endpoint as the default endpoint unless (1) there are only button endpoints, or (2) the first switch endpoint does not contain a supported device type (currently only `DIMMABLE_LIGHT_DEVICE_TYPE_ID`) for a combination button/switch device. I think this solves this issue and it also simplified the logic in `initialize_switch` quite a bit. I added a new test case in the subsequent commit to verify that for a combo button/switch where an unsupported device type is used, the default endpoint is the first button endpoint, the buttons join a MCD profile, and the "unsupported" switch device type joins as a child device.